types/pulsar-client/index.d.ts

```diff
@@ -1,12 +1,12 @@
-// Type definitions for pulsar-client 1.0
+// Type definitions for pulsar-client 1.2
 // Project: https://github.com/apache/pulsar-client-node
 // Definitions by: Brian Walendzinski <https://github.com/bwalendz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 /// <reference types="node" />
 
-export type MessageProperties = unknown[];
+export type MessageProperties = unknown[] | { [key: string ]: string };
```

Comment:
Just use { [key: string]: string }

Reply:
Fixed! I just had to change to **interface** instead of **type**, to please the linter